### PR TITLE
Improve grammar readability

### PR DESCRIPTION
**Pull request by taylorhansen**
Some production rules look like they've been golfed, and others take up more than my screen width, so I decided to reformat the grammar to make it more readable.

### CODE REVIEW

**Comment by taylorhansen** [comment]
Closes #17.

**Review by somebody1234 — APPROVED** [review]
Looks fine but IMO it's slightly overly verbose in some places, and I'd prefer if braces/parens in the grammar use Java style

File: src/vsl/parser/grammar/assignment.ne
Comment (vihanb) [inline src/vsl/parser/grammar/assignment.ne]: This feels like overkill in terms of verbosity. We should use a ternary for this but split it up into multiple lines for readability imo

File: src/vsl/parser/grammar/codeBlock.ne
Comment (vihanb) [inline src/vsl/parser/grammar/codeBlock.ne]: This can be shortened to `id`, and I don't think a 2-letter postprocessing rule should have it's own line 😛

File: src/vsl/parser/grammar/ws.ne
Comment (vihanb) [inline src/vsl/parser/grammar/ws.ne:2]: Does this really need to be on another line? Either way, does this have a trailing newline at the end of file, I can't tell if it's just github being weird from online.

**Review by vihanb — CHANGES_REQUESTED** [review]
Looks good :+1: :D and a lot more clear than the previous grammar but just some minor things.

File: src/vsl/parser/grammar/assignment.ne
Comment (taylorhansen) [reply src/vsl/parser/grammar/assignment.ne]: I could probably use a ternary, but my main argument here is that I'm more concerned about readability than saving a couple lines. We want others to be able to read it, not just ourselves, since it can be very easy to put yourself in a position where you can read your code perfectly but most other developers can't. In that sense, this is a precautionary action I usually take to keep my code less likely to jumble itself up into an unreadable mass of multicolored text.

File: src/vsl/parser/grammar/codeBlock.ne
Comment (taylorhansen) [reply src/vsl/parser/grammar/codeBlock.ne]: Not sure what happened there. I'll fix that.

File: src/vsl/parser/grammar/ws.ne
Comment (taylorhansen) [reply src/vsl/parser/grammar/ws.ne:2]: I wanted to maintain consistency so that's why it's on another line. Regarding your second question, my editor (vim) already handles that newline stuff and GitHub will say when there's no newline at the end via that little red symbol thing when you scroll to the bottom of it.

File: src/vsl/parser/grammar/assignment.ne
Comment (vihanb) [reply src/vsl/parser/grammar/assignment.ne]: Since this is JS I think a lookup-object would be most idiomatic:
```javascript
{
    "let": t.AssignmentType.Constant
    // if we wanted to add more we could just do:
    "ref": t.AssignmentType.Reference
}[data[0]] || t.AssignmentType.Variable
```
as we are likely going to expand the amount of "AssignmentTypes". The reason I'm saying this is that when we expand it an `if..else if..else` is going to be over-verbose for expressing a simple alias. Also the `{ ... }[x] || y` is a pretty common idiom for lookups. You could make that change or if you don't want to you could just merge it and we'll figure it out later

**Review by vihanb — APPROVED** [review]

File: src/vsl/parser/grammar/assignment.ne
Comment (taylorhansen) [reply src/vsl/parser/grammar/assignment.ne]: I'll make this lookup-object thing a const variable above `AssignmentStatement` for now. If you want to you can change it later before or after merging. I don't have write access so you can merge it when you have time.

**Pull request merged**
